staging/src/k8s.io/apiserver/pkg/storage/etcd3/metrics/metrics.go

```diff
@@ -85,6 +86,46 @@ var (
 		},
 		[]string{},
 	)
+	listStorageCount = compbasemetrics.NewCounterVec(
+		&compbasemetrics.CounterOpts{
+			Name:           "apiserver_list_storage_count",
+			Help:           "Number of LIST requests served from storage, split by resource and predicate complexity",
+			StabilityLevel: compbasemetrics.ALPHA,
+		},
+		[]string{"resource", "predicate_complexity"},
+	)
+	listStorageQueries = compbasemetrics.NewCounterVec(
+		&compbasemetrics.CounterOpts{
+			Name:           "apiserver_list_storage_queries",
+			Help:           "Number of storage range queries used to satisfy a LIST request, split by resource",
+			StabilityLevel: compbasemetrics.ALPHA,
+		},
+		[]string{"resource"},
+	)
+	listStorageNumFetched = compbasemetrics.NewCounterVec(
+		&compbasemetrics.CounterOpts{
+			Name:           "apiserver_list_storage_num_fetched",
+			Help:           "Number of objects read from storage in the course of serving a LIST request, split by resource",
+			StabilityLevel: compbasemetrics.ALPHA,
+		},
+		[]string{"resource"},
+	)
+	listStorageNumSelectorEvals = compbasemetrics.NewCounterVec(
+		&compbasemetrics.CounterOpts{
+			Name:           "apiserver_list_storage_num_selector_evals",
+			Help:           "Number of objects tested in the course of serving a LIST request from storage, split by resource and predicate complexity",
+			StabilityLevel: compbasemetrics.ALPHA,
+		},
+		[]string{"resource", "predicate_complexity"},
+	)
+	listStorageNumReturned = compbasemetrics.NewCounterVec(
+		&compbasemetrics.CounterOpts{
```

Comment:
okay, so you want to know the diff of objects fetched and returned, it feel like.. but this is cumulative, so i'm not sure how useful this is going to end up being...